api/src/main/java/com/alibaba/nacos/api/naming/pojo/ServiceInfo.java

```diff
@@ -197,11 +206,16 @@ public static String getKey(String name, String clusters) {
     
     @JsonIgnore
     public String getKeyEncoded() {
+        String serviceName;
         try {
-            return getKey(URLEncoder.encode(name, "UTF-8"), clusters);
+            serviceName = URLEncoder.encode(this.name, "UTF-8");
         } catch (UnsupportedEncodingException e) {
-            return getKey();
+            serviceName = this.name;
+        }
+        if (!StringUtils.isBlank(groupName) && this.name.indexOf(Constants.SERVICE_INFO_SPLITER) == -1) {
```

Comment:
this part codes is duplicate, it can be extract as an method in this class.

Reply:
> this part codes is duplicate, it can be extract as an method in this class.

OK,I will change it